Src/CoyoteRuntime/Values/PrtString.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Plang.PrtSharp.Values
+{
+    [Serializable]
+    public readonly struct PrtString : IPrtValue
+    {
+
+        public bool Equals(PrtString other)
+        {
+            return string.Equals(value, other.value);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            return obj is PrtString other && Equals(other);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public IPrtValue Clone()
+        {
+            return this;
```

Comment:
Shouldnt this be new PrtString(this.val)?

Reply:
since P strings are not mutable sharing a pointer should be fine

Reply:
Agreed!